zookeeper-client/zookeeper-client-c/src/zookeeper.c

```diff
@@ -1324,6 +1324,8 @@ static zhandle_t *zookeeper_init_internal(const char *host, watcher_fn watcher,
     if (cert) {
         zh->fd->cert = calloc(1, sizeof(zcert_t));
         memcpy(zh->fd->cert, cert, sizeof(zcert_t));
+        free(cert->certstr);
```

Comment:
The two statements above copies `cert` from stack to heap. `cert`'s content must still be valid.

Reply:
This [crash `Zookeeper_readOnly::testReadOnlyWithSSL`](https://ci-hadoop.apache.org/blue/organizations/jenkins/zookeeper-precommit-github-pr/detail/PR-2209/2/pipeline/#step-35-log-577) which is the next test of Zookeeper_readOnly::testReadOnly.

Reply:
Thanks for pointing it out, cert has to be freed in zookeeper_close().